### PR TITLE
Generate a dependency report

### PR DESCRIPTION
**Pull request by bsiegel**
Example output is here: https://bl.ocks.org/bsiegel/raw/91377d9b16427cf92725ea8e78e58367/
To produce this example report I downloaded the `packages` artifact from a recent .NET build in our pipeline.

### CODE REVIEW

**Review by mitchdenny — APPROVED** [review]
This looks good. One suggestion for more idiomatic PowerShell in the future is structure your code as a set of Cmdlets which can be used in various ways. For example:

```powershell
.\Get-DependencyIssues.ps1 -Packages Path\**\*.nupkg -LockFile LockFile.props
```

You might then have another script that takes that set of outputs to generate a report or potentially in the future outputs the issues in a format that can be directly picked up by Azure Pipelines. Not something that I would bother leaping into now.

The only other thing I would say - minor nit - is the path. I was hoping that we could keep the paths immediately under ```tools``` as snake case for consistency.

File: tools/AnalyzeDeps/AnalyzeDeps.ps1 [hunk 1]
```diff
@@ -0,0 +1,173 @@
+<#
```
Comment (weshaggard) [inline tools/AnalyzeDeps/AnalyzeDeps.ps1]: We are moving most of these type of things to the eng directory, in fact we plan to delete the tools directory soon so we we will need to move this.

File: tools/AnalyzeDeps/AnalyzeDeps.ps1 [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  } else {
+    $Unlocked.Add($DepName) | Out-Null
+  }
+}
+
+$ExitCode = 0
+if ($Inconsistent) {
+  Write-Warning "$($Inconsistent.Count) inconsistent dependency versions were discovered."
+  # Don't fail the build when inconsistent dependencies are present
+  # TODO: Remove this ASAP
```
Comment (weshaggard) [inline tools/AnalyzeDeps/AnalyzeDeps.ps1]: Is there an issue tracking this TODO?

**Comment by weshaggard** [comment]
@bsiegel I assume you plan to add a step in the Analyze job to produce this report?

File: tools/AnalyzeDeps/AnalyzeDeps.ps1 (hunk 2 above)
Comment (bsiegel) [reply tools/AnalyzeDeps/AnalyzeDeps.ps1]: https://github.com/Azure/azure-sdk-for-net/issues/6771

**Comment by bsiegel** [comment]
> @bsiegel Brandon Siegel FTE I assume you plan to add a step in the Analyze job to produce this report?

@weshaggard It’s all hooked up now

**Pull request merged**